src/backend/distributed/utils/maintenanced.c

```diff
@@ -345,6 +366,42 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 
 
 /*
+ * MetadataTableAccessible returns whether the given metadata table exists
+ * and can be locked using an AccessShareLock.
+ */
+static bool
+MetadataTableAccessible(const char *tableName)
+{
+	bool tableAccessible = false;
+
+	Oid relationId = get_relname_relid(tableName, PG_CATALOG_NAMESPACE);
```

Comment:
> Instead of having RangeVarGetRelidExtended acquire the lock, acquire it in the callback. But there's plenty uglyness around it, so it's not really good.

We figured the same and decided to just do what `RangeVarGetRelidExtended` does manually.

> Hm. That's where you got the above error messages from valgrind?

Yeh, that's an unrelated error being thrown outside of the PG_TRY block, but it seems the longjmp got kind of messed up after the PG_TRY.